pkg/util/k8s_register.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package util
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	k8scsi "k8s.io/api/storage/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
+)
+
+// createOrUpdateCSIDriverInfo Registers CSI driver by creating a CSIDriver object
+func createOrUpdateCSIDriverInfo(csiClientset *kubernetes.Clientset,
+	name string) error {
+	attach := false
+	mountInfo := true
+	// Create CSIDriver object
+	csiDriver := &k8scsi.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: k8scsi.CSIDriverSpec{
+			AttachRequired: &attach,
+			PodInfoOnMount: &mountInfo,
+		},
+	}
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		csidrivers := csiClientset.StorageV1beta1().CSIDrivers()
```

Comment:
I know there is a check in `charts/...` for the kubernetes version, but would this not give an error in case uses deploy manually on an older kubernetes version, or even on a different container platform?

Reply:
this is specific to kubernetes and not enabled by default